pkg/chainsync/chainsync.go

```diff
@@ -114,7 +120,7 @@ func (s *ChainSync) syncHandler(ctx context.Context, peer p2p.Peer, stream p2p.S
 		if err != nil {
 			_ = stream.Reset()
 		} else {
-			_ = stream.Close()
+			_ = stream.FullClose()
```

Comment:
it is not needed to full close here (and i don't think we should use this too much anyway...)

Reply:
this PR aims to standardize our usage of streams. should we change FullClose to Close in all protocols within the scope of this PR in retrieval pushsync etc?

Reply:
so the standardization here (not just in this case) is not necessarily needed and if indeed you want to standardize it might be useful to understand how streams behave. if it is to prevent the case of stream leakage then I argue we should resolve this on the abstraction level

Reply:
my impression is there is practically no difference between the protocols (perhaps with the exception of handshake) on how we use streams. we open them, write and read, and close them at the end or reset on error. this is done mostly consistently in the core protocols. the reason I think it would be nice to have it the same way everywhere is it would decrease uncertainty, and make behaviour more predictable regardless of which context we are in